src/frontend/Typechecker.ml

```diff
@@ -111,17 +111,13 @@ let verify_identifier id : unit =
   then Semantic_error.ident_is_keyword id.id_loc id.name |> error
 
 let distribution_name_variants name =
-  if name = "multiply_log" || name = "binomial_coefficient_log" then [name]
-  else
-    (* this will have some duplicates, but preserves order better *)
-    match Utils.split_distribution_suffix name with
-    | Some (stem, "lpmf") | Some (stem, "lpdf") | Some (stem, "log") ->
-        [name; stem ^ "_lpmf"; stem ^ "_lpdf"; stem ^ "_log"]
-    | Some (stem, "lcdf") | Some (stem, "cdf_log") ->
-        [name; stem ^ "_lcdf"; stem ^ "_cdf_log"]
-    | Some (stem, "lccdf") | Some (stem, "ccdf_log") ->
-        [name; stem ^ "_lccdf"; stem ^ "_ccdf_log"]
-    | _ -> [name]
+  (* this will have some duplicates, but preserves order better *)
+  match Utils.split_distribution_suffix name with
+  | Some (stem, "lpmf") | Some (stem, "lpdf") ->
+      [name; stem ^ "_lpmf"; stem ^ "_lpdf"]
+  | Some (stem, "lcdf") -> [name; stem ^ "_lcdf"]
+  | Some (stem, "lccdf") -> [name; stem ^ "_lccdf"]
+  | _ -> [name]
```

Comment:
We also use it to prevent you from having a `_lpmf` and `_lpdf` of the same name, which seems like a reasonable restriction. 

It does lead to this weird edge case, though:

```stan
model {
  real foo_lpdf;
  real foo_lpmf; // fails
}
```

Reply:
Cleaned up